snuba/querylog/__init__.py

```diff
@@ -12,6 +12,41 @@
 metrics = MetricsWrapper(environment.metrics, "api")
 
 
+def _record_timer_metrics(
+    request: Request, timer: Timer, query_metadata: SnubaQueryMetadata,
+) -> None:
+    final = str(request.query.get_final())
+    referrer = request.referrer or "none"
+    timer.send_metrics_to(
+        metrics,
+        tags={
+            "status": query_metadata.status.value,
+            "referrer": referrer,
+            "parent_api": request.settings.get_parent_api(),
+            "final": final,
+        },
+        mark_tags={
+            "final": final,
+            "referrer": referrer,
+            "parent_api": request.settings.get_parent_api(),
+        },
+    )
+
+
+def _record_attribution_metrics(
+    request: Request, extra_data: Mapping[str, Any]
+) -> None:
+    tags = {
+        "team": request.settings.get_team(),
+        "feature": request.settings.get_feature(),
+        "parent_api": request.settings.get_parent_api(),
+        "referrer": request.referrer,
+    }
+    profile = extra_data["profile"] if extra_data["profile"] else {}
+    bytes_scanned = profile.get("bytes", 0.0)
+    metrics.increment("snuba.attribution", bytes_scanned, tags=tags)
```

Comment:
if there's no profile or bytes scanned, do we still want to record the metric?

Reply:
Yes, because we want to record all the teams/features/referrers even if the numbers are 0. Otherwise there might be some features that we aren't aware of.